File: app/src/main/java/com/hippo/ehviewer/client/parser/GalleryNotAvailableParser.java
package com.hippo.ehviewer.client.parser;

import com.hippo.util.ExceptionUtils;
import com.hippo.util.JsoupUtils;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;

public class GalleryNotAvailableParser {

    public static String parse(String body) {
        String error = null;
        try {
            Document document = Jsoup.parse(body);
            Element d = JsoupUtils.getElementByClass(document, "d");
            //noinspection ConstantConditions
            error = d.child(0).html();
            error = error.replace("<br>", "\n");
        } catch (Throwable e) {
            ExceptionUtils.throwIfFatal(e);
            e.printStackTrace();
        }
        return error;
    }
}

File: app/src/main/java/com/hippo/ehviewer/client/parser/GalleryNotAvailableParser.kt
/*
 * Copyright 2022 Tarsin Norbin
 *
 * This file is part of EhViewer
 *
 * EhViewer is free software: you can redistribute it and/or
 * modify it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * EhViewer is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with EhViewer.
 * If not, see <https://www.gnu.org/licenses/>.
 */

package com.hippo.ehviewer.client.parser;

import com.hippo.util.ExceptionUtils;
import com.hippo.util.JsoupUtils;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;

public class GalleryNotAvailableParser {

    public static String parse(String body) {
        String error = null;
        try {
            Document document = Jsoup.parse(body);
            Element d = JsoupUtils.getElementByClass(document, "d");
            //noinspection ConstantConditions
            error = d.child(0).html();
            error = error.replace("<br>", "\n");
        } catch (Throwable e) {
            ExceptionUtils.throwIfFatal(e);
            e.printStackTrace();
        }
        return error;
    }
}
